src/public/modules/forms/services/spcp-session.client.factory.js

```diff
@@ -16,18 +16,18 @@ function SpcpSession($window, $cookies) {
       SP: 'jwtSp',
       CP: 'jwtCp',
     },
-    setUser: function (authType) {
-      if (session.cookieNames[authType]) {
-        session.cookieName = session.cookieNames[authType]
-        const cookie = $cookies.get(session.cookieName)
-        if (cookie) {
-          const decoded = jwtDecode(cookie)
-          session.userName = decoded.userName
-          session.rememberMe = decoded.rememberMe
-          session.issuedAt = parseInt(decoded.iat)
-          // Every 5 seconds, check if cookie exists and log out if cookie does not exist
-          setInterval(session.checkCookie, 5000)
-        }
+    setUser: function ({ userName, rememberMe, iat, msToExpiry }) {
+      session.userName = userName
+      session.rememberMe = rememberMe
+      session.issuedAt = iat
+      if (!rememberMe) {
+        $timeout(function () {
+          $window.location.reload()
+        }, msToExpiry)
+        // Refresh page after cookie expiry time
+        // Timeout is not set when rememberMe === true because cookie expiry is 30 days
+        // i.e. 2592000000 ms which exceeds the maximum delay value of
```

Comment:
if we returned the date of expiry as suggested in another comment, would it be possible to use `setInterval` (or better still, the angularJS `$interval`) to check if the date of expiry has exceeded?

Reply:
thanks, edited to use $interval